Tests/Test_Simulate_LTC2315.c

```diff
@@ -0,0 +1,223 @@
+            TASK_CANBUS_CONSUMER_PRIO,			// Priority
+            CANBusConsumer_Stk,				// Stack
+            WATERMARK_STACK_LIMIT,	// Watermark limit for debugging
+            TASK_CANBUS_CONSUMER_STACK_SIZE,		// Stack size
+            0,						// Queue size (not needed)
+            10,						// Time quanta (time slice) 10 ticks
+            (void *)0,				// Extension pointer (not needed)
+            OS_OPT_TASK_STK_CHK | OS_OPT_TASK_SAVE_FP,	// Options
+            &err);					// return err code
+    assertOSError(err);
+
+    // Initialize CAN queue
+    CAN_Queue_Init();
+
+	//delete task
+	OSTaskDel(NULL, &p_err); // Delete task
+}
+void trigger(void *p_tmr, void *p_arg) {
+    //tmr_count--;
+    OS_ERR err;
+    if (tmr_count <= 1) {
+        OSTmrDel(&tmr,&err);
+    }
+    Amps_UpdateMeasurements();
+}
+//Task to prevent watchdog from tripping
+void Task2(void *p_arg){
+    OS_ERR err;
+
+    OSTmrCreate(&tmr,
+                "Current Timer",
+                10,
+                10000,
+                OS_OPT_TMR_ONE_SHOT,
+                trigger,
+                NULL,
+                &err);
+    //OSTmrStart(&tmr,&err);
+    
+
+    OSSemPost(&SafetyCheck_Sem4,      //Set semaphore once since Amperes Task doesn't run
+                OS_OPT_POST_1,
+                &err);
+	assertOSError(err);
+    OSSemPost(&SafetyCheck_Sem4,      //Set semaphore once since Battery Balancing Task doesn't run
+                OS_OPT_POST_1,
+                &err);
+	assertOSError(err);
+
+    Amps_UpdateMeasurements();
+    //OSTimeDlyHMSM(0,0,0,50,OS_OPT_TIME_HMSM_STRICT,&err);
+    Amps_UpdateMeasurements();
+
+
+    while(1){
+        OSMutexPend(&WDog_Mutex, 0, OS_OPT_PEND_BLOCKING, NULL, &err);
+        assertOSError(err);
+        if (tmr_count == 0) {
+            Amps_UpdateMeasurements();
+            tmr_count = 10000;
+        }
+        else tmr_count--;
+        WDog_BitMap |= WD_AMPERES;
+        WDog_BitMap |= WD_BALANCING;
+        OSMutexPost(&WDog_Mutex, OS_OPT_POST_NONE, &err);
+        assertOSError(err);
+        //delay of 100ms
+        //Amps_UpdateMeasurements();
+        //BSP_Light_Toggle(RUN);
+    }
+}
+int main(int argc, char **argv) {
+    Simulator_Init(argv[1]);
+    Amps_Init();
+    // if (BSP_WDTimer_DidSystemReset()) {
+    //     Fault_BitMap = Fault_WDOG;
+    //     EnterFaultState();
+    // }
+    OS_ERR err;
+    BSP_PLL_Init();
+
+    //__disable_irq();
+
+    OSInit(&err);
+    assertOSError(err);
+    //while(err != OS_ERR_NONE);
+    OSTaskCreate(&Task1_TCB,"Task 1",Task1,(void *) 0,1,Task1_Stk,16,256,0,0,(void *) 0,OS_OPT_TASK_SAVE_FP | OS_OPT_TASK_STK_CHK,
+                &err);
+                assertOSError(err);
+    OSTaskCreate(&Task2_TCB,"Task 2",Task2,(void *) 0,1,Task2_Stk,16,256,0,0,(void *) 0,OS_OPT_TASK_SAVE_FP | OS_OPT_TASK_STK_CHK,
+                &err);
+                assertOSError(err);
+    //while (err != OS_ERR_NONE);
+    //__enable_irq();
+    OSStart(&err);
+}
```

Comment:
You created a whole test file that redid the hardware. When I meant test file, I basically wanted something that would test just the current simulator (so a Json file or something). You don't have to redo the file, but try making simulator with one of the JSON files and make sure it works (since most of them already have current values in them).